include/awkward/cuda-kernels/cuda_getitem.h

```diff
@@ -0,0 +1,57 @@
+// BSD 3-Clause License; see https://github.com/scikit-hep/awkward-1.0/blob/master/LICENSE
+
+#ifndef AWKWARD_GETITEM_CUH_
+#define AWKWARD_GETITEM_CUH_
```

Comment:
Since these files are inside of a `cuda-kernels` directory, they can have the same names as their `cpu-kernels` counterparts. The directory name is part of the file name (`#include` directives need to include the directory name).

These "only once" directives do have to be globally unique, however. Here, you're using `_CUH` to disambiguate, but there's nothing CUDA about these header files (the interface is pure C). In cpu-kernels, the naming convention is to use an `AWKWARDCPU_` prefix:

https://github.com/scikit-hep/awkward-1.0/blob/62d79bc3b8b2f38c472856e7b884f36b1eb95076/include/awkward/cpu-kernels/getitem.h#L3-L4

so these should just be `AWKWARDCUDA_GETITEM_H`.